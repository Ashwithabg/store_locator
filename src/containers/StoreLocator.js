import React from 'react';
import axios from 'axios';

import Button from "../components/Button";
import Map from "../components/Map";
import Header from '../components/Header';
import mapChooser from "../mapChooser";

import '../App.css';

class StoreLocator extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            currentMap: 'none.png',
            shops: []
        };

        this.chooseMap = this.chooseMap.bind(this);
    }

    async componentDidMount() {
        let response = await axios.get('http://localhost:3000/data/shops.json');
        this.setState({shops: response.data.shops});
    }

    chooseMap(event) {
        this.setState({currentMap: mapChooser(event.target.value)});
    }

    render() {
        let storeButtons = this.state.shops.map((shop, id) => {
            return (<Button handleClick={this.chooseMap} key={id} location={shop.location}/>)
        });

        return (
            <div>
                <Header/>
                <div> {storeButtons} </div>
                <Map imageName={this.state.currentMap} location={this.props.location}/>
            </div>
        )
    }
}


export default StoreLocator;