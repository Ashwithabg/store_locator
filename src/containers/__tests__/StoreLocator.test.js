import React from 'React';
import {shallow} from 'enzyme';
import StoreLocator from '../StoreLocator';

describe('storeLocator', function () {
    let mountedStoreLocator;

    beforeEach(() => {
        mountedStoreLocator = shallow(<StoreLocator/>)
    });

    it('renders Header', function () {
        const mountedHeader = mountedStoreLocator.find('Header');

        expect(mountedHeader.length).toBe(1)
    });

    it('renders two Buttons', function () {
        const mountedButton = mountedStoreLocator.find('Button');

        expect(mountedButton.length).toBe(3)
    });

    it('renders Map', function () {
        const mountedMap = mountedStoreLocator.find('Map');

        expect(mountedMap.length).toBe(1)
    });

    describe('choose map', function () {
        it('should update current map state', function () {
            let locationName = 'testland';
            let locationImageName = locationName + '.png';
            let mountedStoreLocator = shallow(<StoreLocator/>);
            let mockEvent = {target: {value: locationName}};

            mountedStoreLocator.instance().chooseMap(mockEvent);

            expect(mountedStoreLocator.instance().state.currentMap).toBe(locationImageName)

        });
    });

});


