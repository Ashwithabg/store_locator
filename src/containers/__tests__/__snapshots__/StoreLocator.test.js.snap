// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storeLocator renders correctly 1`] = `
<div>
  <div
    className="Header"
  >
    <img
      alt="wired brain logo"
      src="images/wired-brain-coffee-logo.png"
    />
  </div>
  <div>
     
     
  </div>
  <div
    className="MapBox"
  >
    <img
      src="images/none.png"
    />
  </div>
</div>
`;
